### PR TITLE
Use destructures and refactor code for readability

### PR DESCRIPTION
**Pull request by prashant-andani**
Use destructures and refactor code for readability

### CODE REVIEW

File: src/language-vue/parser-vue.js [hunk 1]
```diff
@@ -150,7 +150,11 @@ function parseHTML(html, options) {
       let text;
       let rest;
       let next;
+<<<<<<< pr-fixes
+      
+=======
 
+>>>>>>> master
```
Comment (ikatyang) [inline src/language-vue/parser-vue.js]: It seems you forgot to handle this merge conflict.

File: src/language-graphql/printer-graphql.js [hunk 2]
```diff
@@ -633,12 +633,11 @@ function canAttachComment(node) {
...
-  switch (comment.kind) {
-    case "Comment":
-      return "#" + comment.value.trimRight();
-    default:
-      throw new Error("Not a comment: " + JSON.stringify(comment));
+  if(comment.kind === "Comment") {
+    return "#" + comment.value.trimRight();
+  }
+  else {
+    throw new Error("Not a comment: " + JSON.stringify(comment));
```
Comment (ikatyang) [inline src/language-graphql/printer-graphql.js]: IMO, the original `switch..case` is more readable.

File: src/language-html/embed.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ function embed(path, print, textToDoc, options) {
       // Inline JavaScript
       if (
         parent.type === "script" &&
-        ((!parent.attribs.lang && !parent.attribs.lang) ||
+        ((!parent.attribs.lang) ||
```
Comment (ikatyang) [inline src/language-html/embed.js]: (You have to run `yarn lint --fix` before `git commit`.)

File: src/main/comments.js [hunk 4]
```diff
@@ -294,7 +293,6 @@ function breakTies(tiesToBreak, text, options) {
     return;
   }
   const { precedingNode, followingNode } = tiesToBreak[0];
-
```
Comment (ikatyang) [inline src/main/comments.js]: Any reason you want to remove these empty lines?

File: src/language-html/embed.js (hunk 3 above)
Comment (prashant-andani) [reply src/language-html/embed.js]: Sure !!

File: src/language-vue/parser-vue.js (hunk 1 above)
Comment (prashant-andani) [reply src/language-vue/parser-vue.js]: Oh Yes !! My Bad! :) Fixed

File: src/language-graphql/printer-graphql.js (hunk 2 above)
Comment (prashant-andani) [reply src/language-graphql/printer-graphql.js]: I think if there are only two conditions to be checked upon then "If" and "else" make more sense.

https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04s02.html

**Pull request merged**